Fix incorrect assignment in content-mathml extension.  #1575

dpvc opened the pull request:
Fix incorrect assignment in content-mathml extension, and don't modify original document.

Resolves issue #1575.

zorkow commented on unpacked/extensions/MathML/content-mathml.js:
```diff
@@ -1317,9 +1317,11 @@ MathJax.Extension["MathML/content-mathml"] = (function(HUB) {
+            } else if (arg.nodeName === 'apply' && children.length>2 && children[0].nodeName === 'times' && children[1].nodeName === 'cn' && (n = Number(CToP.getTextContent(children[1]))) < 0) {
               CToP.appendToken(mrow,'mo','\u2212');
-              CToP.getTextContent(children[1]) = -n;// fix me: modifying document
+              arg = arg.cloneNode(true);
```
Looks expensive, but I assume it is necessary to avoid changes to the orig.

dpvc replied:
Yes, that was the reason for it.  I suppose one could change the document, do the transform, then change it back.  Perhaps it doesn't really matter,  since the MathML has been removed from the page anyway, and I think we are working on a copy that has been parsed from the stringified MathML.  Should I just change the document?

zorkow commented:
o/w lgtm.

dpvc commented:
Your comment reminded me that the MathML that content-mathml is processing is actually a temporary MathML tree already, so there is no problem with modifying it, as it is discarded in the end (content-mathml actually replaces it with its result).  So I've removed the cloning and just changed the child directly.
